core/src/main/scala/io/dantb/contentless/FieldType.scala

```
@@ -1,30 +1,102 @@
-      // the val while inside the closure is now considered dangerous and incurs a compile-error under -Wself-implicit
-      eq.eqv(a.itemType, b.itemType) && a.minLength === b.minLength && a.maxLength === b.maxLength
-    case (a, b) => a == b
-  }
-
-  final case class Text(longText: Boolean, validations: Set[Validation])                      extends FieldType
-  final case class Media(mimeTypeGroup: Set[MimeTypeGroup])                                   extends FieldType
-  final case class Reference(linkContentTypes: Set[ContentTypeId])                            extends FieldType
-  final case class Array(itemType: FieldType, minLength: Option[Int], maxLength: Option[Int]) extends FieldType
-  final case class RichText(validations: Set[Validation])                                     extends FieldType
-  case object Integer                                                                         extends FieldType
-  case object Number                                                                          extends FieldType
-  case object Boolean                                                                         extends FieldType
-  case object Json                                                                            extends FieldType
-  case object DateTime                                                                        extends FieldType
-  case object Location                                                                        extends FieldType
+  given eq: Eq[FieldType] = Eq.fromUniversalEquals
+
+  final case class Text(
+      longText: Boolean,
+      charBounds: Option[Validation.Size],
+      allowedValues: Option[NonEmptyList[String]],
+      matchesRegex: Option[Regexp]
+  ) extends FieldType:
+    def validations: Set[Validation] =
+      charBounds.toSet ++ matchesRegex.toSet ++ allowedValues.map(Validation.ContainedIn(_)).toSet
+  object Text:
+    def fromValidations(longText: Boolean, vs: Set[Validation]): Text =
```

How would this API be used? Is it ok to silently ignore validations that aren't pertinent?

---

Hmm yeah it's a bit gross having one overarching `Validation` ADT, really just a legacy artefact... I've added an issue to have field-specific ones so I don't forget